src/test/java/info/sjd/controller/ItemControllerTest.java

```diff
@@ -0,0 +1,184 @@
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(itemService, times(1)).getAll();
+    }
+
+    @Test
+    void getByCodeTest() throws URISyntaxException {
+        Item item = Item.builder().
+                name("test_name").
+                code("t_code").
+                price(123).
+                availability(1).
+                build();
+        when(itemService.getByCode(anyString())).thenReturn(Collections.singletonList(item));
+        UriComponents uriComponents = UriComponentsBuilder.newInstance()
+                .host("/item")
+                .path("by-code").query("code={keyword}").buildAndExpand("code");
+        RequestEntity<Item> requestEntity = new RequestEntity<>(HttpMethod.GET, new URI(uriComponents.toString()));
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(itemService, times(1)).getByCode(anyString());
+    }
+
+    @Test
+    void getAllByCartTest() throws URISyntaxException {
+        Item item = Item.builder().
+                name("test_name").
+                code("t_code").
+                price(123).
+                availability(1).
+                build();
+        when(itemService.getAllByCart(anyInt())).thenReturn(Collections.singletonList(item));
+        UriComponents uriComponents = UriComponentsBuilder.newInstance()
+                .host("/item")
+                .path("cart_id").query("id={keyword}").buildAndExpand("23");
+        RequestEntity<Item> requestEntity = new RequestEntity<>(HttpMethod.GET, new URI(uriComponents.toString()));
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(itemService, times(1)).getAllByCart(anyInt());
+    }
+
+    @Test
+    void getAllAvailableTest() throws URISyntaxException {
+        Item item = Item.builder().
+                name("test_name").
+                code("t_code").
+                price(123).
+                availability(1).
+                build();
+        when(itemService.getAllAvailable()).thenReturn(Collections.singletonList(item));
+        UriComponents uriComponents = UriComponentsBuilder.newInstance()
+                .host("/item")
+                .path("all-available").build();
+        RequestEntity<Item> requestEntity = new RequestEntity<>(HttpMethod.GET, new URI(uriComponents.toString()));
+        ResponseEntity<List> responseEntity = testRestTemplate.exchange(requestEntity, List.class);
+        assertEquals(responseEntity.getStatusCode(), HttpStatus.OK);
+        verify(itemService, times(1)).getAllAvailable();
+    }
+
+    @Test
+    void deleteTest() throws URISyntaxException {
```

Comment:
You must mock delete method too.
Do not forget check its invocation.

Reply:
done